cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -183,22 +183,6 @@ liborc::RowReaderOptions default_row_reader_options() {
   return options;
 }
 
-// Proactively check the availability of timezone database.
-// Remove it once https://issues.apache.org/jira/browse/ORC-1661 has been fixed.
-Status check_timezone_database_availability() {
-  auto tz_dir = std::getenv("TZDIR");
-  bool is_tzdb_avaiable = tz_dir != nullptr
```

Comment:
I would personally keep this check, as mentioned earlier (https://github.com/apache/arrow/pull/40697#discussion_r1535231818), I think it's important to provide am informative error message. And with this check, we can easily improve the clarity of the error message a lot.

And as mentioned here, we can limit this to ORC < 2.0, to use ORC's better handling when it is available.

Reply:
Well, the question is also how likely it is to use an outdated version of ORC C++ with a new version of PyArrow. Assuming people only use binary builds from PyPI or conda-forge, they should get a new ORC as well.

Reply:
Yes, if we ensure to update our ORC dependency before 16.0, I suppose that's fine as well, but AFAIK at the moment the wheels include ORC 1.9

Reply:
@wgtmac mentioned here https://github.com/apache/arrow/issues/36026#issuecomment-2006161559 that this is blocked by Java, but for the wheel building that shouldn't matter. So it's probably a matter of updating the vcpkg snapshot version? (last time we did this -> https://github.com/apache/arrow/pull/39622)

Reply:
There is no reason that ORC Java should block ORC C++ in any case, is there?

Reply:
That said, for now vcpkg still has 1.9.2:
https://github.com/microsoft/vcpkg/blob/master/ports/orc/vcpkg.json

Reply:
What about upgrading ORC to v2.0.0 in this patch as well? Do we still need to add the check in this case?

Reply:
Finally I added back the TZDB check if the ORC version is under 2.0.0. PTAL @pitrou @jorisvandenbossche